### PR TITLE
add removeMarkers to leaflet and openlayers

### PR DESCRIPTION
**Pull request by PeterTheOne**
add removeMarkers functions to leaflet and openlayers (googlemaps3 already has this function) as noted in #117.

This is required for https://github.com/SemanticMediaWiki/SemanticMaps/pull/52 to work with leaflet and openlayers.

### CODE REVIEW

File: includes/services/OpenLayers/jquery.openlayers.js [hunk 1]
```diff
@@ -64,45 +66,56 @@
...
+				data:markerData
+			});
+			curLayer.addMarker(marker); // Create and add the marker.
+		};
+
+		this.removeMarkers = function () {
+			var map = this.map;
+			$.each(this.groupLayers, function(index, layer) {
+				map.removeLayer(layer);
+			});
```
Comment (JeroenDeDauw) [inline includes/services/OpenLayers/jquery.openlayers.js]: Does this not also remove things that are not makers? Such as images and whatnot? And "base layers"? Perhaps the later are not removed, and perhaps the former should be counted as images (or rather as what this function is actually about). I'm very fuzzy on the stuff, though someone should check.

**Comment by JeroenDeDauw** [comment]
Generally looks good to me. Did you test that the changes don't cause obvious breakage? (Yay for not having any tests that cover this stuff at all)

File: includes/services/OpenLayers/jquery.openlayers.js (hunk 1 above)
Comment (PeterTheOne) [inline includes/services/OpenLayers/jquery.openlayers.js]: First I wanted to remove markers directly but, some layers also need to be removed. The `removeMarkers` function only removes `groupLayers` they are only used for markers. Lines, etc. have their own layers like `lineLayer`.

**Comment by PeterTheOne** [comment]
I did test my use case and it did not break, but I haven't tested with shapes like lines, etc.

**Pull request merged**